File: apps/mobile/lib/models/issue.dart
class IssueMin {
  const IssueMin({
    required this.id,
    required this.issueKey,
    required this.issueType,
    required this.summary,
    required this.status,
    required this.priority,
    this.storyPoints,
    required this.backlogRank,
    this.assigneeName,
  });

  final int id;
  final String issueKey;
  final String issueType;
  final String summary;
  final String status;
  final String priority;
  final int? storyPoints;
  final int backlogRank;
  final String? assigneeName;

  factory IssueMin.fromJson(Map<String, dynamic> j) {
    return IssueMin(
      id: (j['id'] as num).toInt(),
      issueKey: j['issueKey'] as String,
      issueType: j['issueType'] as String,
      summary: j['summary'] as String,
      status: j['status'] as String,
      priority: j['priority'] as String,
      storyPoints: (j['storyPoints'] as num?)?.toInt(),
      backlogRank: (j['backlogRank'] as num?)?.toInt() ?? 0,
      assigneeName: j['assigneeName'] as String?,
    );
  }
}

class IssueLabelItem {
  const IssueLabelItem({required this.id, required this.name});
  final int id;
  final String name;
  factory IssueLabelItem.fromJson(Map<String, dynamic> j) {
    return IssueLabelItem(
      id: (j['id'] as num).toInt(),
      name: j['name'] as String,
    );
  }
}

class IssueComponentItem {
  const IssueComponentItem({required this.id, required this.name});
  final int id;
  final String name;
  factory IssueComponentItem.fromJson(Map<String, dynamic> j) {
    return IssueComponentItem(
      id: (j['id'] as num).toInt(),
      name: j['name'] as String,
    );
  }
}

class IssueDetail {
  const IssueDetail({
    required this.id,
    required this.issueKey,
    required this.projectId,
    required this.projectKey,
    required this.issueType,
    required this.summary,
    this.description,
    required this.status,
    required this.priority,
    this.storyPoints,
    this.assigneeId,
    this.assigneeName,
    required this.reporterId,
    this.reporterName,
    this.parentId,
    this.parentKey,
    this.sprintId,
    required this.backlogRank,
    this.securityLevel,
    required this.createdAt,
    required this.updatedAt,
    required this.labels,
    required this.components,
  });

  final int id;
  final String issueKey;
  final int projectId;
  final String projectKey;
  final String issueType;
  final String summary;
  final String? description;
  final String status;
  final String priority;
  final int? storyPoints;
  final int? assigneeId;
  final String? assigneeName;
  final int reporterId;
  final String? reporterName;
  final int? parentId;
  final String? parentKey;
  final int? sprintId;
  final int backlogRank;
  final String? securityLevel;
  final String createdAt;
  final String updatedAt;
  final List<IssueLabelItem> labels;
  final List<IssueComponentItem> components;

  factory IssueDetail.fromJson(Map<String, dynamic> j) {
    return IssueDetail(
      id: (j['id'] as num).toInt(),
      issueKey: j['issueKey'] as String,
      projectId: (j['projectId'] as num).toInt(),
      projectKey: j['projectKey'] as String,
      issueType: j['issueType'] as String,
      summary: j['summary'] as String,
      description: j['description'] as String?,
      status: j['status'] as String,
      priority: j['priority'] as String,
      storyPoints: (j['storyPoints'] as num?)?.toInt(),
      assigneeId: (j['assigneeId'] as num?)?.toInt(),
      assigneeName: j['assigneeName'] as String?,
      reporterId: (j['reporterId'] as num).toInt(),
      reporterName: j['reporterName'] as String?,
      parentId: (j['parentId'] as num?)?.toInt(),
      parentKey: j['parentKey'] as String?,
      sprintId: (j['sprintId'] as num?)?.toInt(),
      backlogRank: (j['backlogRank'] as num?)?.toInt() ?? 0,
      securityLevel: j['securityLevel'] as String?,
      createdAt: j['createdAt'] as String,
      updatedAt: j['updatedAt'] as String,
      labels: (j['labels'] as List<dynamic>? ?? [])
          .map((e) => IssueLabelItem.fromJson(e as Map<String, dynamic>))
          .toList(),
      components: (j['components'] as List<dynamic>? ?? [])
          .map((e) => IssueComponentItem.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }
}

class SpringPage<T> {
  const SpringPage({
    required this.content,
    required this.totalElements,
    required this.totalPages,
    required this.size,
    required this.number,
  });

  final List<T> content;
  final int totalElements;
  final int totalPages;
  final int size;
  final int number;

  factory SpringPage.fromJson(
    Map<String, dynamic> j,
    T Function(Map<String, dynamic>) item,
  ) {
    final raw = j['content'] as List<dynamic>? ?? [];
    return SpringPage<T>(
      content: raw.map((e) => item(e as Map<String, dynamic>)).toList(),
      totalElements: (j['totalElements'] as num?)?.toInt() ?? 0,
      totalPages: (j['totalPages'] as num?)?.toInt() ?? 0,
      size: (j['size'] as num?)?.toInt() ?? 0,
      number: (j['number'] as num?)?.toInt() ?? 0,
    );
  }
}

File: src/main/java/com/pch/mng/issue/Issue.java
package com.pch.mng.issue;

import com.pch.mng.global.enums.IssueStatus;
import com.pch.mng.global.enums.IssueType;
import com.pch.mng.global.enums.Priority;
import com.pch.mng.global.enums.SecurityLevel;
import com.pch.mng.project.Project;
import com.pch.mng.sprint.Sprint;
import com.pch.mng.user.UserAccount;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@NoArgsConstructor
@Data
@Entity
@Table(name = "issue_tb")
public class Issue {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String issueKey;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id", nullable = false)
    private Project project;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IssueType issueType;

    @Column(nullable = false)
    private String summary;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IssueStatus status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Priority priority;

    private Integer storyPoints;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assignee_id")
    private UserAccount assignee;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reporter_id", nullable = false)
    private UserAccount reporter;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Issue parent;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sprint_id")
    private Sprint sprint;

    /** 스프린트 미배정(backlog) 이슈 정렬용. 스프린트에 넣은 이슈는 0으로 둘 수 있다. */
    @Column(nullable = false)
    private long backlogRank;

    @Enumerated(EnumType.STRING)
    private SecurityLevel securityLevel;

    /** Epic 전용 로드맵 기간 (FR-012). 비-Epic는 null 유지. */
    @Column(name = "epic_start_date")
    private LocalDate epicStartDate;

    @Column(name = "epic_end_date")
    private LocalDate epicEndDate;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @Builder
    public Issue(String issueKey, Project project, IssueType issueType, String summary,
                 String description, IssueStatus status, Priority priority, Integer storyPoints,
                 UserAccount assignee, UserAccount reporter, Issue parent, Sprint sprint,
                 long backlogRank, SecurityLevel securityLevel,
                 LocalDate epicStartDate, LocalDate epicEndDate) {
        this.issueKey = issueKey;
        this.project = project;
        this.issueType = issueType;
        this.summary = summary;
        this.description = description;
        this.status = status;
        this.priority = priority;
        this.storyPoints = storyPoints;
        this.assignee = assignee;
        this.reporter = reporter;
        this.parent = parent;
        this.sprint = sprint;
        this.backlogRank = backlogRank;
        this.securityLevel = securityLevel;
        this.epicStartDate = epicStartDate;
        this.epicEndDate = epicEndDate;
    }
}
